### ISSUE TITLE
Kickstart a project kotlin-server

### ISSUE THREAD

**Issue by jmfayard** [LouisCAD/kotlin-libraries-playground#24]
Currently all samples are inside `kotlin-jvm` and are self-contained in their own package, with each having its `main()` function.

But what about servers?

Ideally I would like to have a new directory/project `kotlin-server` that will contain an implementation of the same simple server spec in ktor but also later spring boot, http4k, javalin or whatever people want to contribute.

For a real project, one would use a multimodule Gradle build for that, but for pedagogy simplicity is key so if that's possible to put all implementations in a single module, that's better!

**Comment by jaguililla** [comment]
Would it be safe to use a single Gradle project for many different web servers? Couldn't that create dependency problems (different versions)?

**Comment by LouisCAD** [comment]
@jaguililla Do you have any example where a dependency upgrade is not possible for a Kotlin/JVM server framework?

**Comment by jaguililla** [comment]
No, but it would be hard to tell without knowing the list of examples in advance... And being it open, it could eventually generate a problem with conflicting dependencies.

And also, there are products sensible to the classpath (Spring).

Wouldn't you consider creating a multimodule Gradle project? It wouldn't be much work and it would be safer.

**Comment by LouisCAD** [comment]
We are considering making a multi-modules project, yes.
We need to get it right to allow easy code sharing, while keeping it simple for contributors.
I think we'll make one shared module, plus one module per framework as an entry point.

**Comment by jaguililla** [comment]
If you share more details, I could be interested in doing it. What should a typical sample implement?

Or you could take the TechEmpower Benchmark implementations as valid examples :) https://github.com/TechEmpower/FrameworkBenchmarks/tree/master/frameworks/Kotlin

**Comment by jmfayard** [comment]
I am giving a try, will start with a Spring Boot module

**Comment by jmfayard** [comment]
@jaguililla @fultonbrown @NikkyAI
Here is my draft server implementation done with spring boot
https://github.com/LouisCAD/kotlin-libraries-playground/pull/93

If you have any feedback, you are welcome.

Everyone could then reimplement it with $FAVORITE_FRAMEWORK
like Spring-Fu Ktor Micronaut Quarkus Vertx Http4k Javalin
